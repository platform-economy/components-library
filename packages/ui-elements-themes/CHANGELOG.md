# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.2](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.1...@relayr/ui-elements-themes@0.1.2) (2019-08-22)

**Note:** Version bump only for package @relayr/ui-elements-themes






## [0.1.1](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.0...@relayr/ui-elements-themes@0.1.1) (2019-08-21)

**Note:** Version bump only for package @relayr/ui-elements-themes






# 0.1.0 (2019-08-02)


### Features

* **ui-elements-basic:** new InputLabel component ([abb6076](https://github.com/relayr/ui-elements/commit/abb6076))
* **ui-elements-layout:** add Card component ([e8db0cf](https://github.com/relayr/ui-elements/commit/e8db0cf))
* **ui-elements-tables:** Table components ([#102](https://github.com/relayr/ui-elements/issues/102)) ([b1b8d6f](https://github.com/relayr/ui-elements/commit/b1b8d6f))
* **ui-elements-themes:** custom font linking system ([8e9f0f0](https://github.com/relayr/ui-elements/commit/8e9f0f0))
