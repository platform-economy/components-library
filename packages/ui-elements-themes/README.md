# ui-elements-themes

## Add to project

via npm:

```bash
npm install --save @relayr/ui-elements-themes
```

or Yarn:
```bash
yarn add @relayr/ui-elements-themes
```
