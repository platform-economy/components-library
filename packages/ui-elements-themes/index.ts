export * from './themes/light';
export * from './themes/dark';
export * from './themes/test';
export * from './themes/_common/paletteDark';
export * from './themes/_common/paletteLight';
export * from './themes/_common/methods';
export * from './themes/_common/fonts';
export * from './themes/_common/withTestTheme';
