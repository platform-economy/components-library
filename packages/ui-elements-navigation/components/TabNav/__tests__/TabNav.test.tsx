import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import 'jest-styled-components';

import { TabNav } from '../index';
import { withTestTheme } from '@relayr/ui-elements-themes';

enzyme.configure({ adapter: new Adapter() });

describe('ui-elements-navigation', () => {
  describe('Components', () => {
    describe('TabNav', () => {
      it('should render with two children with one of them active', () => {
        const tree = TestRenderer
          .create(withTestTheme(
            <TabNav>
              <TabNav.Tab active={true}>Test Tab</TabNav.Tab>
              <TabNav.Tab>Test Tab 2</TabNav.Tab>
            </TabNav>))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('should properly handle onClick', () => {
        const clickHandler = jest.fn();
        const tree = enzyme.mount(withTestTheme(
          <TabNav>
            <TabNav.Tab onClick={clickHandler}>Click Tab Test</TabNav.Tab>
          </TabNav>));
        tree.find('li').simulate('click');
        expect(clickHandler).toHaveBeenCalledTimes(1);
      });
    });
  });
});
