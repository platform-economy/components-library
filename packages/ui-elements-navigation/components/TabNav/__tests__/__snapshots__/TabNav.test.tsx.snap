// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-navigation Components TabNav should render with two children with one of them active 1`] = `
.c1 {
  color: #text;
  min-height: 1.25em;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  -webkit-letter-spacing: -0.38px;
  -moz-letter-spacing: -0.38px;
  -ms-letter-spacing: -0.38px;
  letter-spacing: -0.38px;
  text-align: center;
  padding: 10px 45px;
  cursor: pointer;
  border-bottom: 3px solid transparent;
}

.c1.active {
  font-weight: bold;
  color: #active_lowered;
}

.c1.disabled {
  color: #inactive;
  cursor: not-allowed;
}

.c1:hover:not(.disabled):not(.active) {
  color: #active_hovered;
  border-bottom: 3px solid #active_hovered;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: auto;
  padding: 0;
  list-style-type: none;
  border-bottom: 1px solid #foreground;
}

.c0:after {
  content: '';
  border-bottom: 3px solid #active_lowered;
  position: absolute;
  width: 0px;
  top: -3px;
  left: 0px;
  -webkit-transition: 0.2s ease-out;
  transition: 0.2s ease-out;
}

<ul
  className="c0"
>
  <li
    className="active c1"
  >
    Test Tab
  </li>
  <li
    className="c1"
  >
    Test Tab 2
  </li>
</ul>
`;
